### PR TITLE
29 search bar component

### PR DESCRIPTION
**Pull request by ArthiRagothaman**

### CODE REVIEW

File: src/components/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+										<span
+											className={`block truncate ${
+												selected
+													? "font-medium"
+													: "font-normal"
+											}`}
+										>
+											{list.name}
+										</span>
+										{selected ? (
```
Comment (NoHara42) [inline src/components/SearchBar.tsx:77]: no need for ternary here, use the `&&` operator.
see:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND#short-circuit_evaluation

File: src/components/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+					</div>
+
+					<Combobox.Options className=" ui-absolute mt-1 max-h-60 w-full overflow-auto rounded-md bg-white py-1 text-base">
+						{filteredItems.map((list) => (
+							<Combobox.Option
+								key={list.id}
+								value={list}
+								className={({ active }) =>
+									`relative cursor-default select-none py-2 pl-10 pr-4 ${
+										active ? "bg-teal-600 text-white" : " "
```
Comment (NoHara42) [inline src/components/SearchBar.tsx:62]: see below comment

**Review by NoHara42 — CHANGES_REQUESTED** [review]
please remove lock files and merge main into your branch before asking for a review @ArthiRagothaman

File: src/components/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+					<Combobox.Options className=" ui-absolute mt-1 max-h-60 w-full overflow-auto rounded-md bg-secondary-transparent py-1 from-text-white_transparent">
+						{filteredItems.map((list) => (
+							<Combobox.Option
+								key={list.id}
+								value={list}
+								className={({ active }) =>
+									`relative cursor-default select-none py-2 pl-10 pr-4 ${
+										active
+											? " bg-primary-default-Solid text-text-white"
+											: " "
```
Comment (NoHara42) [inline src/components/SearchBar.tsx:60]: looks like a use-case for `&&` , see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND#short-circuit_evaluation

**Review by NoHara42 — CHANGES_REQUESTED** [review]
see changes, also the dropdown list of options doesn't match what design has requested, the list of options is not in-line with the searchbar
@ArthiRagothaman 
image

**Comment by ArthiRagothaman** [comment]
As requested updated the changes

**Review by NoHara42 — CHANGES_REQUESTED** [review]
comment from previous review not resolved, see edge case in screenshot below
- (x) text goes under icon when long enough
- (x) storybook does not have disabled variant, nor disabled control for the disabled state seen in the mockup

image

**Comment by ArthiRagothaman** [comment]
- input contains padding so that the text does not go over the icon when it is long enough
1.  Even after removed the padding it didn't work. Yes, it will go over the icon. The issue will get resolved only if the icon changed to left side.
(Drop down list restricts to choose one of the options for user so no need to type so long)

- storybook has disabled variant, with disabled control for the disabled state seen in the mockup

2. This is not my component. There is separate home component for those disable variant [home](https://www.figma.com/file/MwSsIVfA5KQIZCdhIcASnC/Grocery-App?node-id=65%3A132&t=O6xM2HbvfXx8xfGj-4)

**Comment by ArthiRagothaman** [comment]
- [ ]  Now Text will not overlap the icon seems working
- [ ]  updated the styles 
- [ ]  Not clear about the disabled variant

**Review by NoHara42 — CHANGES_REQUESTED** [review]
heroicons have not been added to this branches `package.json` file as a dependency.

Therefore I unfortunately cannot test this branch

**Comment by ArthiRagothaman** [comment]
Now icons will update as per [Update icons](https://github.com/devhausleipzig/camp8-final-project/issues/119) this link.
I have included feather icons in my component.

**Pull request merged**